scipy/stats/tests/test_contingency.py

```diff
@@ -196,3 +196,28 @@ def test_chi2_contingency_bad_args():
     obs = np.empty((0, 8))
     assert_raises(ValueError, chi2_contingency, obs)
 
+
+def test_bad_association_args():
+    # Invalid Test Statistic
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], "X")
+    # Invalid array shape
+    assert_raises(ValueError, association, [[[1, 2]], [[3, 4]]], "cramer")
+    # Invalid array shape (asymmetric)
+    assert_raises(ValueError, association, [[1, 2], [3, 4], [5]], 'cramer')
+    # chi2_contingency exception
+    assert_raises(Exception, association, [[-1, 10], [1, 2]], 'cramer')
+    # Invalid array (asymmetric)
+    assert_raises(ValueError, association, [[1, 2], [3]], 'cramer')
+    # Invalid array item value type
+
+
+def test_assoc():
+    # 2d Array
+    obs1 = np.array([[12, 13, 14, 15, 16],
+                     [17, 16, 18, 19, 11],
+                     [9, 15, 14, 12, 11]])
+    stats = ['cramer', 'tschuprow', 'pearson']
+    results = [0.09222412010290792, 0.0775509319944633, 0.12932925727138758]
+    for stat in stats:
+        a = association(observed=obs1, method=stat)
+        assert_allclose(a, results[stats.index(stat)])
```

Comment:
Here's the preferred way to do this, using [`pytest.mark.parametrize`](https://docs.pytest.org/en/stable/parametrize.html).  First, add the line `import pytest` to the imports at the beginning of the file.  Then write this test as
```suggestion
@pytest.mark.parametrize('stat, expected',
                         [('cramer', 0.09222412010290792),
                          ('tschuprow', 0.0775509319944633),
                          ('pearson', 0.12932925727138758)])
def test_assoc(stat, expected):
    # 2d Array
    obs1 = np.array([[12, 13, 14, 15, 16],
                     [17, 16, 18, 19, 11],
                     [9, 15, 14, 12, 11]])
    a = association(observed=obs1, method=stat)
    assert_allclose(a, expected)
```